hazelcast-client/src/test/java/com/hazelcast/client/spi/impl/ClientInvocationServiceImplTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.spi.impl;
+
+import com.hazelcast.client.connection.nio.ClientConnection;
+import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
+import com.hazelcast.client.impl.protocol.codec.ClientPingCodec;
+import com.hazelcast.client.test.ClientTestSupport;
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientInvocationServiceImplTest extends ClientTestSupport {
+
+    private final TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
+    private HazelcastClientInstanceImpl client;
+
+    @After
+    public void cleanup() {
+        hazelcastFactory.terminateAll();
+    }
+
+    @Before
+    public void setUp() {
+        hazelcastFactory.newHazelcastInstance();
+        client = getHazelcastClientInstanceImpl(hazelcastFactory.newHazelcastClient());
+    }
+
+    @Test(expected = IOException.class)
+    public void testCleanResourcesTask_rejectsPendingInvocationsWithClosedConnections() throws Throwable {
+        ClientConnection closedConn = closedConnection();
+        ClientInvocation invocation = new ClientInvocation(client, ClientPingCodec.encodeRequest(), null, closedConn);
+
+        ClientInvocationFuture future = invocation.invoke();
+        try {
+            future.get();
```

Comment:
This test seems to complete with IOException even when sending the invocation packet with "Packet not sent" without `CleanResourcesTask` involvement.

Reply:
@ihsandemir Yes, you are right. I have modified the test code, can you check again ?